packages/webapp/src/components/Filter/FilterMultiSelect/index.tsx

```diff
@@ -54,6 +54,8 @@ export const FilterMultiSelect = ({
         defaultFilterState[option.value].active = true;
       }
     });
+
+    onChange?.(filterRef.current![filterKey]);
   }, [value]);
```

Comment:
Changing the value passed to this `onChange` should also now make the filter multi-select compatible with Report Filters; however for the purpose of using in animals (`useVisibleBreeds` makes use of the current `filterRef`) the change in the timing is the only thing that matters.

Reply:
Not 100% sure but I think this might be solvable by getting rid of this `useEffect` altogether and calling things in the right order in the `onChange` function:

```jsx
const handleChange = (value: ComponentFilterOption[]): void => {
    setValue(value);
    filterRef.current![filterKey] = produce(defaultFilterState, (defaultFilterState) => {
      for (const option of value) {
        defaultFilterState[option.value].active = true;
      }
    });
    onChange?.(value as unknown as FilterState);
    /* NOTE: What is being passed here is in ComponentFilterOption format, and would not work if the state-setting onChange had been passed. However, the finance report filters -- the only container using a state-setting onChange -- don't use multi-select filters. */
  };

  return (
    <ReactSelect
      //@ts-ignore
      style={style} // I suspect the forwardRef is at fault for the type error?
      placeholder={`${t('common:SHOWING_ALL')}...`}
      options={options}
      label={subject}
      value={value}
      onChange={handleChange}
      isMulti
      isSearchable
      className={className}
    />
  );
```

Long term, I feel pretty strongly that we shouldn't be using refs but a state variable to store the filter's state (I'm not sure why we ended up with a ref), so the less we have to rely on that ref the easier it'll be to get rid of it.

Reply:
Thanks!!! Although in testing that change I realized I was missing a `useEffect()` in `useVisibleBreeds`...that needs the `filterRef` 😅 

That's interesting because personally I love the `filterRef` -- I think it's tied to how I dislike the React render cycle, but the idea a value that _just is_ independent of the component's render cycle is such a relief 😂  But it's use here is probably tied how the component uses a different data format than the Redux state (source of so much mess and complexity) -- and I bet they could both be eliminated together.